### PR TITLE
Wrap DappRequestSwitchCustomChainCoordinatorDelegate and DappRequestSwitchExistingChainCoordinatorDelegate to reduce in coordinator size #4061

### LINKED ISSUE #4061: Wrap DappRequestSwitchCustomChainCoordinatorDelegate and DappRequestSwitchExistingChainCoordinatorDelegate to reduce in coordinator size

### PR DESCRIPTION
**Pull request by oa-s**
Closes #4061

### CODE REVIEW

File: AlphaWallet/Core/Helpers/DappRequestHandler.swift [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+        private func processRestartQueueAndRestartUI() {
+            delegate?.processRestartQueueAndRestartUI()
+        }
+    }
+}
+extension InCoordinator.DappRequestHandler: DappRequestSwitchCustomChainCoordinatorDelegate {
+
+    func notifySuccessful(withCallbackId callbackId: SwitchCustomChainCallbackId, inCoordinator coordinator: DappRequestSwitchCustomChainCoordinator) {
+        switch callbackId {
```
Comment (hboon) [inline AlphaWallet/Core/Helpers/DappRequestHandler.swift:36]: Feels like such code switch (to differentiate WalletConnect and Dapp) here and all the functions following this suggests we should have some more refactoring (not necessarily removing this switch-case code).

**Review by hboon — APPROVED** [review]

File: AlphaWallet/Core/Helpers/DappRequestHandler.swift (hunk 1 above)
Comment (oa-s) [reply AlphaWallet/Core/Helpers/DappRequestHandler.swift:36]: Not clearly sure that understood an idea, if we remove some of `switch callbackId {` checks the code will be exexuted for both wallet connect and dapp,
 and that might cause some issues with sending wrong responses

Comment (hboon) [reply AlphaWallet/Core/Helpers/DappRequestHandler.swift:36]: Right.

**Comment by hboon** [comment]
@oa-s remember to fix tests and merge.

**Pull request merged**